ckan/logic/action/update.py

```diff
@@ -1308,3 +1311,87 @@ def bulk_update_delete(context, data_dict):
 
     _check_access('bulk_update_delete', context, data_dict)
     _bulk_update_dataset(context, data_dict, {'state': 'deleted'})
+
+
+def config_option_update(context, data_dict):
+    '''
+
+    .. versionadded:: 2.4
+
+    Modify the values of one or more CKAN configuration options
+
+    It takes arbitrary key, value pairs and checks the keys against the
+    config options update schema. If some of the provided keys are not present
+    in the schema a :py:class:`~ckan.plugins.logic.ValidationError` is raised.
+    The values are then validated against the schema, and if validation is
+    passed each key, value config option:
+
+    * It is stored on the ``system_info`` database table
+    * The Pylons ``config`` object is updated.
+    * The ``app_globals`` (``g``) object is updated (this only happens for
+      options explicitly defined in the ``app_globals`` module.
+
+    The following lists a ``key`` parameter, but this should be replaced by
+    whichever config options want to be updated, eg::
+
+        get_action('config_option_update)({}, {
+            'ckan.site_title': 'My Open Data site',
+            'ckan.homepage_layout': 2,
+        })
+
```

Comment:
How about adding a warning like this: You should only add config options that you know in the code are ok to be updated during runtime, such as ones you've added in your own extension, or have reviewed the use of in core ckan.